lib/systems/src/controllers/DrivebrainController.cpp

```diff
@@ -1,72 +1,82 @@
 
-    if( (curr_millis - last_speed_setpoint_timestamp) > _worst_message_latencies.worst_speed_setpoint_latency_so_far)
-    {
-        _worst_message_latencies.worst_speed_setpoint_latency_so_far = static_cast<int64_t>(curr_millis - last_speed_setpoint_timestamp);   
-    } else if((curr_millis - last_torque_lim_timestamp) > _worst_message_latencies.worst_torque_lim_latency_so_far)
-    {
-        _worst_message_latencies.worst_torque_lim_latency_so_far = static_cast<int64_t>(curr_millis - last_torque_lim_timestamp);   
-    }
-
-    bool timing_failure = (speed_setpoint_msg_too_latent || torque_limit_message_too_latent || not_all_messages_recvd || latency_diff_too_high);
-
-    // if _timing_failure is true and we want to re-init db-controller, connection between the drive brain AND button must be pressed
-    bool rq_db_controller = state.interface_data.dash_input_state.data_btn_is_pressed;
-
-    if (rq_db_controller && (!timing_failure))
-    {
-        // timing failure should be false here
-        _timing_failure = false;
+    if (drivebrain_reinit_button_pressed && (!_should_run_controller)) {
+        _should_run_controller = true;
     }
 
     DrivetrainCommand_s output;
-    if (!timing_failure && (!_timing_failure))
-    {   
-        output = db_input.get_command();
-    }
-    else
-    {
-        _timing_failure = true;
+
+    if (_should_run_controller && !_telem_latency_info.timing_failure) {
+        output = db_telem_input.get_command();
+    } else if (_should_run_controller && !_aux_latency_info.timing_failure) {
+        output = db_auxillary_input.get_command();
+    } else {
+        _should_run_controller = false;
         DrivetrainCommand_s coast_to_stop = {
             .desired_speeds = {0.0f, 0.0f, 0.0f, 0.0f},
-            .torque_limits = {0.0f, 0.0f, 0.0f, 0.0f}
         };
         output = coast_to_stop;
     }
+
+    // Handle worst latency updates
+    if (_last_reset_worse_latency_clock == 0) {
+        _last_reset_worse_latency_clock = curr_millis;
+    }
+
+    if (curr_millis - _last_reset_worse_latency_clock > WORST_LATENCY_PERIOD_MS) {
+        _last_reset_worse_latency_clock = curr_millis; 
+        _telem_latency_info.worst_period_millis = 0;
+        _aux_latency_info.worst_period_millis = 0;
+    }
+
+    unsigned long aux_latency_millis = std::max(
+        curr_millis - db_auxillary_input.desired_speeds.last_recv_millis, 
+        curr_millis - db_auxillary_input.torque_limits.last_recv_millis
+    );
+
+    unsigned long telem_latency_millis = std::max(
+        curr_millis - db_telem_input.desired_speeds.last_recv_millis, 
+        curr_millis - db_telem_input.torque_limits.last_recv_millis
+    );
+
+    _aux_latency_info.worst_period_millis = std::max(_aux_latency_info.worst_period_millis, aux_latency_millis);
+    _telem_latency_info.worst_period_millis = std::max(_aux_latency_info.worst_period_millis, telem_latency_millis);
```

Comment:
you took the max of `_aux_latency_info.worst_period_millis` not `_telem_latency_info.worst_period_millis`

Reply:
good catch